pkg/kubelet/kubelet.go

```diff
@@ -2967,6 +3013,14 @@ func (kl *Kubelet) handlePodResourcesResize(pod *v1.Pod, podStatus *kubecontaine
 			// If a resize is in progress, make sure the cache has the correct state in case the Kubelet restarted.
 			kl.statusManager.SetPodResizeInProgressCondition(pod.UID, "", "", false)
 		} else {
+			// If pod resize completed, print pod resize complete event
+			if kl.statusManager.IsPodResizeStatusInProgress(allocatedPod.UID) {
```

Comment:
I think my comment in https://github.com/kubernetes/kubernetes/pull/130387#discussion_r2066952376 is still relevant - we should be emitting an event when either the PodResizeInProgress condition or the PodResizePending condition is cleared. The latter happens when a deferred or infeasible resize is reverted.

This may be slightly trickier to reason about, but maybe you can do the following:
1. At the beginning of `handlePodResourcesResize`, store the original `kl.statusManager.GetPodResizeConditions()`
2. At the end of the defer block, if the original conditions were not empty, but `kl.statusManager.GetPodResizeConditions()` is now empty, emit the event

If you do it this way, I don't think we need `kl.statusManager.IsPodResizeStatusInProgress` after all (sorry)

cc @tallclair